### DISCUSSION TITLE
Is there any advantage of Mariadb backend over Sqlite3?

### DISCUSSION THREAD

**Discussion by DistractedCanuck** [dani-garcia/vaultwarden#1617] [General]
Hi, 

I'm running a homelab with Bitwarden_rs for myself and my close friends and family. Probably at most 15 users.  I'm wondering if there would be an advantage to running Bitwarden with a Mariadb backend or if I'm good using the default sqlite3.  I'm looking at setting up backups using duplicati and either backblaze or wasabi.  (maybe google drive.... :S) 

I also have seafile and photoprism that currently use Mariadb.  They are both running with independent Mariadb containers.  Would I be benefitted to try to combine all three? or should I keep it simple and just backup the one sqlit3 file for bitwarden and dump the other two independently for backup? 

Any opinion is helpful.  I'm just getting into the world of selfhosting and want to try to set up best practices now when theres essentially no data in my db to lose or migrate rather than pulling my hair out later trying to back track.  

Thanks so much!

**Comment by jjlin** [comment]
I don’t think there’s usually a need to use MySQL or PostgreSQL for bitwarden_rs. Those databases may be better for write-intensive workloads, but a password manager doesn’t really do that many writes since people don’t tend to be constantly updating their password entries. The SQLite backend is also the most widely used and tested, so you’re less likely to run into weird corner cases. Also, bitwarden_rs often performs a large number of small queries (especially when syncing a user's password database), which SQLite is better-suited to since everything is usually local.

However, I wouldn’t recommend using SQLite on a network file system like NFS or Samba/CIFS, or on something like a Raspberry Pi without at least a “high endurance” SD card. So, if the local storage on your compute node has questionable reliability, that could be a good reason to run with MySQL or PostgreSQL on a remote server that has better storage. File attachments (if you use them) aren’t stored in the database though, so those could still be an issue.

**Reply by DistractedCanuck** [reply]
That makes a lot of sense.  Thank you for the clear and detailed explanation.  I'm running everything on an old laptop with a Crucial CT525MX3 512GB in it.  It's advertised to 220TB of Endurance for whatever that number is worth.  I've been using it since 2016.  But as a drive in my windows laptop for uni.  So I'd expect it to still have quite a bit of life left in it hopefully. Though it was pretty full a lot.  (Knock on wood, but yay backups!)
